### ISSUE TITLE
Plan architectural update for undermidi in view of midiserver changes

### ISSUE THREAD

**Issue by oubiwann** [ut-proj/undermidi#54]
Questions to entertain:
* How would the project change is we wanted to support both the Go midiserver and the Erlang NIF midiserver?
* How might things be different if we supported arbitrary backends?
* What if we completely remove midiserver or anything else like it from the deps, and just keep the gen_servers in this project?

Tasks:
* [ ] ❌ Devise a first version of switchable backend support (config-based? at runtime?)
* [ ] ❌ In `src/undermidi/supervisor.lfe`, change `go-server` to `backend`
* [ ] ❌ Move code in `src/undermidi/go`into midiserver-go (the fork that will be created once the NIF epic is merged into midiserver`); this will mean moving that project's current code into a `go` subdirectory and creating a new `erlang` subdir that will be a full (rebar-based) Erlang project
* [x] Evaluate code changes in midiserver epic branch and determine how much of that can move into undermidi itself
* [x] Continue work in undermidi
